### ISSUE TITLE
EVReflection handling of members which are generics

### ISSUE THREAD

**Issue by vronin** [evermeer/EVReflection#33]
I know that generics needs to be handled special way.

However, workaround only shows how to handle when the whole object (which is serialized/deserialized is Generic). 

It's not clear what to do when a member is of such generic class

Following code will fail

```
public class MyGenericBase: EVObject {
    var test : String = ""
}

public class InstanceObject: EVObject {
    var name:String?
}   

public class MyGenericObject<T where T:NSObject>: MyGenericBase, EVGenericsKVC {
    var data: T = T()
    var array: [T] = [T]()

    required public init() {
        super.init()
    }

    public override func setValue(value: AnyObject!, forUndefinedKey key: String) {
        switch key {
        case "data":
            data = value as! T
        case "array":
            array = value as! [T]
        default:
            print("---> setValue '\(value)' for key '\(key)' should be handled.")
        }
    }
}

class Foo {
   var bar : MyGenericObject<InstanceObject> = MyGenericObject<InstanceObject>()
}

var foo = Foo()
foo.bar.data.name = "Test"

var json = foo.toJsonString()
var foo2 = Foo(json: json) // This will fail
```

The reason for a failure is swiftClassTypeFromString.  NSClassFromString returns nil for Generic type.

Two questions:
1) Am I missing another workaround to handle this situation
2) Is it possible to do automatical handling for this case?

**Comment by evermeer** [comment]
The first thing I notice  is that the Foo class should have NSObject of EVObject as it's base class. But since you are able to do the .toJsonString() I think that one is lost in the copy / paste to this issue. I just added your test case to EVReflection. It does not fail on that line. In my case foo2 is created. I did rename the Foo object since I already had a Foo object. There is indeed an issue with this. The json will show the Test value for the name, but the new object won't. I will investigate that now.
